bpf-gpl/tc.c

```diff
@@ -784,6 +784,23 @@ static CALI_BPF_INLINE struct fwd calico_tc_skb_accepted(struct cali_tc_ctx *ctx
 				CALI_DEBUG("rt found for 0x%x local %d\n",
 						bpf_ntohl(state->post_nat_ip_dst), !!cali_rt_is_local(rt));
 
+				if (cali_rt_is_workload(rt) && state->tun_ip == 0) {
```

Comment:
Need to check if the rt is to a _local_ workload?

Reply:
yeah, this was taken a bit blindly from the ARP branch, should be for local only. And in fact, we want to do that also on the node where it was forwarded to.